module/executiondatasync/execution_data/execution_data.go

```diff
@@ -20,10 +20,49 @@ type ChunkExecutionData struct {
 	// TrieUpdate is the trie update generated by executing the collection
 	// This includes a list of all registers updated during the execution
 	TrieUpdate *ledger.TrieUpdate
+
+	// TransactionResults are the results of executing the transactions in the collection
+	// This includes all of the data from flow.TransactionResult, except uses a boolean
+	// value to indicate if an error occurred instead of a full error message.
+	TransactionResults []TransactionResult
 }
 
 // BlockExecutionData represents the execution data of a block.
 type BlockExecutionData struct {
 	BlockID             flow.Identifier
 	ChunkExecutionDatas []*ChunkExecutionData
 }
+
+// TransactionResult represents a flow.TransactionResult that's encoded within a ChunkExecutionData.
+//
+// The primary difference is this version includes a boolean value indicated if the transaction's
+// execution resulted in an error, instead of the full error message. Error messages are prone to
+// non-determinism due to (undiscovered) bugs in the execution and verification code. Rather than
+// including the error and risking execution forks if an undetected bug is introduced, we simplify
+// it to just a boolean value.
+type TransactionResult struct {
+	// TransactionID is the ID of the transaction this result was emitted from.
+	TransactionID flow.Identifier
+	// HasError is true if the transaction's execution resulted in an error, false otherwise.
+	HasError bool
```

Comment:
What about succeed?
```suggestion
	Succeed bool
```

Reply:
I was worried that these are semantically different.

All we can see at the protocol level is whether or not a transaction's execution resulted in an error. However, completing without an error doesn't necessarily mean it succeeded in the eyes of the sender, just that it didn't fail.

Reply:
ok, makes sense

Reply:
Not sure, but should this be `HadError`... Or maybe `Failed`. Probably should be past tense since execution already happened.

Reply:
switched it to `Failed` in [8982ad6](https://github.com/onflow/flow-go/pull/4699/commits/8982ad65ca02c79f48c4bfd9407c1a46e2d46f28)